grails-app/services/org/pih/warehouse/inventory/StockMovementService.groovy

```diff
@@ -1132,106 +1140,438 @@ class StockMovementService {
         }
     }
 
-    void validatePicklistListImport(StockMovement stockMovement, List<PickPageItem> pickPageItems, List<ImportPickCommand> picklistItems) {
-        Map<String, List> picklistItemsGroupedByRequisitionItem = picklistItems.groupBy { it.id }
+    /**
+     * This method needs to be broken down into multiple smaller methods.
+     *
+     * @param command
+     */
+    void validatePicklistListImport(PicklistImportDataCommand command) {
 
-        picklistItems?.each { ImportPickCommand data ->
-            data.validate()
-            // skip validation if id is empty since mos tof the validation relies on an existing requisition id
-            if (!data.id) {
-                return
-            }
-            PickPageItem pickPageItem = pickPageItems.find { it.requisitionItem?.id == data.id }
-            if (!pickPageItem) {
-                data.errors.rejectValue(
-                        "id",
-                        "importPickCommand.requisitionItem.notFound.error",
-                        [data.id] as Object[],
-                        "Requisition item id: ${data.id} not found"
-                )
-            }
-            if (pickPageItem) {
-                AvailableItem availableItem = pickPageItem.getAvailableItem(data.binLocation, data.lotNumber)
-                ApplicationTagLib g = grailsApplication.mainContext.getBean(ApplicationTagLib.class)
+        StockMovement stockMovement = command.stockMovement
 
-                if (!availableItem) {
-                    String lotNumberName = data.lotNumber ?: g.message(code: "default.noLotNumber.label", default: Constants.DEFAULT_LOT_NUMBER)
-                    String binLocationName = data.binLocation ?: g.message(code: "default.noBinLocation.label", default: Constants.DEFAULT_BIN_LOCATION_NAME)
-                    data.errors.rejectValue(
-                            "id",
-                            "importPickCommand.availableItem.notFound.error",
-                            [lotNumberName, binLocationName] as Object[],
-                            "There is no item available with: lot \"${lotNumberName}\" and bin location \"${binLocationName}\""
-                    )
-                }
+        // The parent requisition items that picklist items are associated with
+        List<PickPageItem> pickPageItems = command.pickPageItems
+
+        // FIXME Remove this if it's no longer being used
+        // Configuration property that allows overpick for a given location
+        Boolean supportsOverpick = command.location.supports(ActivityCode.ALLOW_OVERPICK)
```

Comment:
like I mentioned in the comment before, this should probably be checked for origin location and not current logged-in.